### PR TITLE
XIVY-15134 cardinality and mappedBy options

### PR DESCRIPTION
**Pull request by ivy-lgi**
Use new endpoints to set options for cardinality and mappedBy selections.

### CODE REVIEW

File: playwright/tests/global.teardown.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { rm } from 'node:fs';
 
 const teardown = async () => {
-  rm('./tests/integration/projects/dataclass-test-project/dataclasses/temp', { force: true, recursive: true }, () => {});
+  rm('./dataclass-test-project/dataclasses/temp', { force: true, recursive: true }, () => {});
```
Comment (ivy-lli) [inline playwright/tests/global.teardown.ts:4]: ah sorry 🙈

File: playwright/tests/pageobjects/abstract/Select.ts [hunk 2]
```diff
@@ -13,8 +13,17 @@ export class Select {
     }
   }
 
-  async choose(value: string) {
+  async choose(value?: string) {
+    if (value === undefined) {
+      return;
+    }
```
Comment (ivy-lli) [inline playwright/tests/pageobjects/abstract/Select.ts:19]: why needed to have this optional?

File: playwright/tests/pageobjects/field/entity/FieldAssociation.ts [hunk 3]
```diff
@@ -62,11 +61,11 @@ export class FieldAssociation {
     }
   }
 
-  async fillValues(cardinality: string, cascadeTypes: FieldAssociationCascadeTypes, mappedBy: string, removeOrphans: boolean) {
+  async fillValues(cardinality: string, cascadeTypes: FieldAssociationCascadeTypes, mappedBy: string | undefined, removeOrphans: boolean) {
     await this.collapsible.open();
     await this.cardinality.choose(cardinality);
     await this.fillCascadeTypes(cascadeTypes);
-    await this.mappedBy.locator.fill(mappedBy);
+    await this.mappedBy.choose(mappedBy);
```
Comment (ivy-lli) [inline playwright/tests/pageobjects/field/entity/FieldAssociation.ts:68]: maybe better to check if undefined here than add this complexity to the select page object?

File: packages/protocol/package-lock.json
Comment (ivy-lli) [inline packages/protocol/package-lock.json:1]: once run a `npm install` inside of this package?
remove this file :)

**Review by ivy-lli — APPROVED** [review]
good, some hints:

File: playwright/tests/pageobjects/field/entity/FieldAssociation.ts (hunk 3 above)
Comment (ivy-lgi) [reply playwright/tests/pageobjects/field/entity/FieldAssociation.ts:68]: Yes, makes much more sense.
The difference was between selecting the `emptyItem` (`choose('')`) and not making a change to the selection (`choose(undefined)`).

**Review by ivy-lli — APPROVED** [review]

**Pull request merged**